File: 백준/Silver/2217. 로프/로프.py
import sys 
input = sys.stdin.readline
sys.setrecursionlimit
n = int(input())
ropes = [ 0 for _ in range(n)]
for i in range(n):
    ropes[i]=int(input())
ropes.sort(reverse=True)
for i in range(n):
    ropes[i] = ropes[i] * (i + 1)
print(max(ropes))

File: 백준/Silver/2217. 로프/로프.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	var n int
	fmt.Fscanf(reader, "%d\n", &n)
	var ropes = make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Fscanf(reader, "%d\n", &ropes[i])
	}
	sort.Slice(ropes, func(i, j int) bool {
		return ropes[i] > ropes[j]
	})
	for i := 0; i < n; i++ {
		ropes[i] = ropes[i] * (i + 1)
	}
	sort.Ints(ropes)
	fmt.Println(ropes[len(ropes)-1])
}
